app/components/sidebars/main/threads_entry/index.tsx

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect} from 'react';
+import {TouchableHighlight, Text, View} from 'react-native';
+import {injectIntl, intlShape} from 'react-intl';
+import {useDispatch, useSelector} from 'react-redux';
+
+import {handleViewingGlobalThreadsScreen} from '@actions/views/threads';
+import {NavigationTypes} from '@constants';
+import CompassIcon from '@components/compass_icon';
+import {getStyleSheet} from '@components/sidebars/main/channels_list/channel_item/channel_item';
+
+import {getThreads} from '@mm-redux/actions/threads';
+import {getTheme} from '@mm-redux/selectors/entities/preferences';
+import {getCurrentTeamId} from '@mm-redux/selectors/entities/teams';
+import {getTeamThreadCounts} from '@mm-redux/selectors/entities/threads';
+import {getCurrentUserId} from '@mm-redux/selectors/entities/users';
+import type {Theme} from '@mm-redux/types/preferences';
+import type {GlobalState} from '@mm-redux/types/store';
+import EventEmitter from '@mm-redux/utils/event_emitter';
+import {getViewingGlobalThreads, getViewingGlobalThreadsUnread} from '@selectors/threads';
+import {preventDoubleTap} from '@utils/tap';
+import {changeOpacity} from '@utils/theme';
+import {makeStyleFromTheme} from '@mm-redux/utils/theme_utils';
+import Badge from '@components/badge';
+
+type Props = {
+    intl: typeof intlShape;
+};
+
+const ThreadsEntry = ({intl}: Props) => {
+    const viewingGlobalThreads = useSelector(getViewingGlobalThreads);
```

Comment:
no need to do it now, but I want to see this changed in a later PR where we get rid of `useSelector` this will cause massive pain when porting to Gekidou

Reply:
Missed this one while refactoring global thread components. Fixed it now. Thank you.